src/client.rs

```diff
@@ -34,21 +35,57 @@ pub struct Client {
 #[builder(setter(into))]
 #[builder(name = "ClientBuilder")]
 pub struct ClientBuilderInternal {
+    /// Set an optional Github token.
+    /// This allows for more requests before
+    /// getting rate-limited.
     github_token: Option<String>,
+    /// Check links matching this set of regular expressions
     includes: Option<RegexSet>,
-    excludes: Excludes,
+    /// Exclude links matching this set of regular expressions
+    excludes: Option<RegexSet>,
+    /// Exclude all private network addresses
+    exclude_all_private: bool,
+    /// Exclude private IP addresses
+    exclude_private_ips: bool,
+    /// Exclude link-local IPs
+    exclude_link_local_ips: bool,
+    /// Exclude loopback IP addresses (e.g. 127.0.0.1)
+    exclude_loopback_ips: bool,
```

Comment:
I'm in two minds about it.

1:

We ask the users to provide complex objects from other crates like `custom_headers: HeaderMap` or `accepted: Option<HashSet<http::StatusCode>>` -- yet we "explode" the `Excludes` which should be part of the public API.

I get the "removing coupling" argument, but maybe the functionality of producing `Excludes` from config could be moved to config, and then we could have a nice public constructor/builder for `Excludes` what would go into the library's public API?

2:

Accepting just primitive types makes the builder much simpler and straightforward to use.

Honestly, don't know what's the right choice here, but I'm happy to give the current approach a try.

Reply:
I can feel you. The one good thing about the builder is that not all options have to be configured at once. I assume that many users would simply pass `exclude_all_private(true)` and perhaps a regex pattern if needed. Probably many wouldn't even touch the options and go with the defaults, which is great. 😄 

The other thing I like about "unrolling" the config options like this is that the IDE can provide useful completions. E.g. if a user isn't sure what exclude they might need, they might simply type `exclude` and see what the IDE suggests, whereas they'd have to import the ExcludeBuilder and learn how to configure it before passing it to the ClientBuilder, which could break their train of thought.

On the other hand your argument about having other complex types is true. It was a direct translation from the previous constructor, but one could argue that it's time to break that up and make life a little easier for the user by providing an `add_header()` and an `accept()` method and simply creating the more complex types during the build.
Would that make sense?

Reply:
> make life a little easier for the user by providing an add_header() and an accept() method and simply creating the more complex types during the build.
Would that make sense?

Definitely! I'm very much a fan of this approach. This would be the simplest API from the human-friendliness perspective. And if we had both `add_header(&str, &str)` and `add_headers(HeaderMap)` that would cover all the use cases.